File: libs/utils/lib-anim/src/main/java/com/yyxnb/what/anim/action/AnimAction.java
package com.yyxnb.what.anim.action;

import com.yyxnb.what.anim.R;

/**
 * 动画样式
 *
 * @author yyx
 */
public interface AnimAction {

    /**
     * 默认动画效果
     */
    int ANIM_DEFAULT = -1;

    /**
     * 没有动画效果
     */
    int ANIM_EMPTY = 0;

    /**
     * 缩放+淡入淡出动画
     */
    int ANIM_SCALE = R.style.Animation_Scale;

    /**
     * 吐司动画
     */
    int ANIM_TOAST = android.R.style.Animation_Toast;

    /**
     * 上进上出
     */
    int ANIM_TOP = R.style.Animation_TopToTop;

    /**
     * 下进下出
     */
    int ANIM_BOTTOM = R.style.Animation_BottomToBottom;

    /**
     * 左进左出
     */
    int ANIM_LEFT = R.style.Animation_LeftToLeft;

    /**
     * 右进右出
     */
    int ANIM_RIGHT = R.style.Animation_RightToRight;

    /**
     * 左进右出
     */
    int ANIM_LEFT_TO_RIGHT = R.style.Animation_LeftToRight;

    /**
     * 右进左出
     */
    int ANIM_RIGHT_TO_LEFT = R.style.Animation_RightToLeft;
}

File: widget/lib_common/src/main/java/com/yyxnb/common/action/AnimAction.kt
package com.yyxnb.common.action

import com.yyxnb.common.R

/**
 * 动画样式
 *
 * @author yyx
 */
interface AnimAction {
    companion object {
        /**
         * 默认动画效果
         */
        const val ANIM_DEFAULT = -1

        /**
         * 没有动画效果
         */
        const val ANIM_EMPTY = 0

        /**
         * 缩放+淡入淡出动画
         */
        val ANIM_SCALE: Int = R.style.Animation_Scale

        /**
         * 吐司动画
         */
        const val ANIM_TOAST = android.R.style.Animation_Toast

        /**
         * 上进上出
         */
        val ANIM_TOP: Int = R.style.Animation_TopToTop

        /**
         * 下进下出
         */
        val ANIM_BOTTOM: Int = R.style.Animation_BottomToBottom

        /**
         * 左进左出
         */
        val ANIM_LEFT: Int = R.style.Animation_LeftToLeft

        /**
         * 右进右出
         */
        val ANIM_RIGHT: Int = R.style.Animation_RightToRight

        /**
         * 左进右出
         */
        val ANIM_LEFT_TO_RIGHT: Int = R.style.Animation_LeftToRight

        /**
         * 右进左出
         */
        val ANIM_RIGHT_TO_LEFT: Int = R.style.Animation_RightToLeft
    }
}